### ISSUE TITLE
Brickstore crashes silently on click action

### ISSUE THREAD

**Issue by xaraxarax** [rgriebl/brickstore#587]
WIndows 10 19044.1889, Brickstore newer 2022.4.1

(more or less) reproducable, regardless if document new or opened: 

- right or left click click on any line
- open add dialog, somtimes the dialog show short up
- close settings dialog sometimes

add:

after i went back to versio 2022.4.1, i noticed that the first right click on any line also takes longer, maybe it's just a timing problem ...

https://user-images.githubusercontent.com/79638214/190234617-9461275a-828a-44f3-8c75-67c195842c68.mp4

As a workaround I'll revert to 2022.4.1, but any advice is appreciated

**Comment by rgriebl** [comment]
Works perfectly fine for me. I think all these problems are graphics driver related. Whenever you click an item, you should immediately see the part info, but you don't. Please check 3 things:
1) do you have "crash reporting" enabled in the settings? I can't find any crash report that would match your problem
2) Test the newest build 774 from here: https://github.com/rgriebl/brickstore/actions/runs/3054699124  (this also uses a newer Qt 3D renderer.
3) Update the drivers of your graphics card (or integrated GPU).

**Comment by xaraxarax** [comment]
Has worked perfectly for me too, until today

1. now yes, but it looks like it's not sending anything, at least not visibly
2. It was exact the 744. 
3. No, i think i'll stick with 2022.4.1 until brickstore, qt or any other component wants to play with my graphics card again ;)

**Comment by rgriebl** [comment]
I guess you are the guy with the integrated Intel HD 3000 that generated 8 sentry crash reports in the last 20 minutes :)
And it's exactly as expected: you are using an 11 year old GPU that was buggy from day 1. You are not alone with this problem: google for "OpenGL HD 3000 problem" and you get ~3 million results.

The "latest" driver (if you could call it that) is from 2015: https://www.intel.com/content/www/us/en/download/18388/intel-graphics-driver-for-windows-10-15-40-4th-gen.html

You can either try to install the newer driver or you'll have to stick with BrickStore 2022.4.1 forever. Nobody will add support for this mess of a GPU after all these years.

Closing as there is nothing more that I could do.

**Comment by xaraxarax** [comment]
Thanks for the explanation, I see what you mean now. Never had problems with the graphics card before, that's why I'm skeptical about your suggestion. I'm probably on the verge of dying out.

**Comment by rgriebl** [comment]
I hate to drop support for old hardware, but the hardware world is too fast-paced to make it possible, when a giant company like Intel only supports their integrated GPU (with sales in the millions) for just 4 years.

**Comment by outbreaker8** [comment]
Hi. Probably having similar problems. After the "real graphic card" died, I continued with the onboard GPU. Which obviously is an HD Graphics 2000.
Same problem for this model as well?

**Comment by rgriebl** [comment]
Most likely. Please try the latest driver from here: https://www.intel.com/content/www/us/en/download/17608/24970/intel-graphics-driver-for-windows-15-28.html

If the new driver doesn't help, I might look into somehow disabling the 3D part rendering for those old Intel HD 2000/3000.

**Comment by outbreaker8** [comment]
Tried to install the driver you mentioned, but installation routine warned, that the installed driver is newer (at least from 2016 🤣)
![image](https://user-images.githubusercontent.com/17637836/190875228-6b32e6db-3a01-4930-9be8-eeec4d2770c4.png)

https://www.intel.com/content/www/us/en/support/intel-driver-support-assistant.html
-> didn't find something to install

I personally never used the 3D rendering. So disabling it would be no issue.

**Comment by rgriebl** [comment]
Thanks for testing. I think I can assemble a GPU blacklist from the sentry crash reports.

**Comment by rgriebl** [comment]
Didn't expect so many GPUs from the Stone Age without DirectX 11 and broken OpenGL drivers still being in use... the leader right now is a NVIDIA Geforce 210 from 2009 😉 

        "Microsoft Basic Render Driver",
        "Intel(R) HD Graphics",
        "Intel(R) HD Graphics 3000",
        "NVIDIA GeForce 210",
        "NVIDIA GeForce GT 525M",
        "AMD Radeon HD 8240",

**Comment by rgriebl** [comment]
Can you both please test this build: https://github.com/rgriebl/brickstore/actions/runs/3074997149
This should block the problematic 3D rendering on a bunch of old GPUs, including the HD and HD3000.

**Comment by xaraxarax** [comment]
Works better but it still crashes. But unfortunately not reproducible.

https://user-images.githubusercontent.com/79638214/190880390-6c501125-8dcd-45e2-abd7-fc5a2dbebd2d.mp4

**Comment by xaraxarax** [comment]
Moment, but reproducible, it is e.g. 15439. Very strange...

https://user-images.githubusercontent.com/79638214/190880520-a68018ba-307e-40f4-8393-70f12566cd08.mp4

will test further tomorrow

**Comment by rgriebl** [comment]
Seems like the 3D renderer was initialized ... the moment you clicked an item that has a 3D model, it crashed.
The sentry report shows the exact same crash as before. No idea why the blacklisting didn't work: if I fake my GPU id string, it works for me.

**Comment by rgriebl** [comment]
Checked the GPU id before it was initialized... this one should do the trick: 
https://github.com/rgriebl/brickstore/actions/runs/3075500050

**Comment by outbreaker8** [comment]
installed #582 -> works ("3D" grayed out)
🙏 nice!

**Comment by xaraxarax** [comment]
Works great, brickstore is playing along with my gpu again :D I've never used the ldraw/3d function either, so I don't miss anything.

Big thanks for that!

P.S. Do you have an explanation why the first click, right or left doesn't matter, always takes longer? What's happening in the background?

**Comment by xaraxarax** [comment]
Thanks for the explanation. You know that I have an old box, you know all about it. But you don't shoot your dog just because it's getting old. I think the ticket can be closed. It's still stable.

**Comment by rgriebl** [comment]
tr("Nicht alles, was hinkt, ist ein Vergleich") 😉 
Sure it's *your* beloved dog, but I don't want to spend a lot of *my* free time supporting *your* old dog.

Still, crashes are not acceptable. So I debugged the problem in Qt with my colleagues and we found the culprit (the ambient occlusion render step doesn't check if the GPU actually has support for it). Hopefully I can drop the blacklist again with the next Qt update...
